shape.cc

```diff
@@ -56,12 +56,12 @@ arglist_t shp_args[] = {
 static void
 my_rd_init(const QString& fname)
 {
-  ihandle = SHPOpen(qPrintable(fname), "rb");
+  ihandle = SHPOpen(fname.toUtf8(), "rb");
```

Comment:
This will break support for non-utf8 unix, and it will not fix windows in cases where windows fopen would not work.  Under normal circumstances these calls lead to fopen.  To fix windows it seems necessary to call SASetupUtf8Hooks at least after SHPopen, DBFOpen, SHPcreate, DBFCreate, DBFCreateEx, DBFCloneEmpty (any where that leads to SASetupDefaultHooks being called in shapelib) to get windows utf8 support.  SASetupUtf8Hooks can only be called if SHPAPI_UTF8_HOOKS is defined, and it will only work if SHPAPI_WINDOWS is also defined.
We also have *NO* test cases.  However it should be relatively easy to see if filenames like test_encoding_fileαβγ.shx and test_encoding_fileαβγ.shp at least find files on input and write them on output.
This comment applies to all changes in this file.